pkg/drivers/kic/kic.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kic
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/state"
+	"github.com/pkg/errors"
+	pkgdrivers "k8s.io/minikube/pkg/drivers"
+	"k8s.io/minikube/pkg/drivers/kic/node"
+	"k8s.io/minikube/pkg/minikube/command"
+	"k8s.io/minikube/pkg/minikube/pkg/drivers/kic/cri"
+)
+
+// https://minikube.sigs.k8s.io/docs/reference/drivers/kic/
+type Driver struct {
+	*drivers.BaseDriver
+	*pkgdrivers.CommonDriver
+	URL           string
+	exec          command.Runner
+	OciBinary     string
+	ImageSha      string
+	CPU           int
+	Memory        int
+	APIServerPort int32
+}
+
+// Config is configuration for the kic driver
+type Config struct {
+	MachineName   string
+	CPU           int
+	Memory        int
+	StorePath     string
+	OciBinary     string // oci tool to use (docker, podman,...)
+	ImageSha      string // image name with sha to use for the node
```

Comment:
Is SHA a necessary detail to know for an implementer? If so, what kind of SHA? 256?

Alternatively, what do you think about `ImageDigest`?